Dockerfile

```diff
@@ -1,30 +1,41 @@
-# setup build image
-FROM golang:1.11 AS build
+# Minify client side assets (JavaScript)
+FROM node:latest AS build-js
 
-# build Gophish binary
-WORKDIR /build/gophish
+WORKDIR /build
 COPY . .
-RUN go get -d -v ./...
-RUN go build
 
+RUN npm install gulp gulp-cli -g
+RUN npm install --only=dev
+RUN gulp
 
-# setup run image
-FROM debian:stable-slim
 
-RUN apt-get update && \
-    apt-get install --no-install-recommends -y \
-    jq && \
-    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+# Build Golang binary
+FROM golang:1.11 AS build-golang
+
+WORKDIR /go/src/github.com/gophish/gophish
+COPY . .
+
+RUN go get -v && go build -v
 
-# copy Gophish assets from the build image
-WORKDIR /gophish
-COPY --from=build /build/gophish/ /gophish/
-RUN chmod +x gophish
 
-# expose the admin port to the host
+# Runtime container
+FROM debian:stable-slim
+
+WORKDIR /opt/gophish
+RUN useradd -d /opt/gophish -s /bin/bash app
+
+RUN apt-get update && \
+	apt-get install --no-install-recommends -y jq && \
+	apt-get clean && \
+	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+COPY --from=build-golang /go/src/github.com/gophish/gophish/ ./
+COPY --from=build-js /build/static/js/dist/ ./static/js/dist/
+COPY --from=build-js /build/static/css/dist/ ./static/css/dist/
+RUN touch config.json.tmp && chown app. config.json config.json.tmp
```

Comment:
Once the `USER app` line is moved up, you can remove the `&& chown...` line since it won't be needed anymore.